src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -1116,21 +1117,30 @@ private static FunctionResult LoadContentResultBuilder(ObjectParser objectParser
             return new(ErrorType.Create(errorDiagnostic));
         }
 
-          private static FunctionResult ReadEnvironmentVariableResultBuilder (IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, FunctionCallSyntaxBase functionCall, ImmutableArray<TypeSymbol> argumentTypes)
-         {
+        private static FunctionResult ReadEnvironmentVariableResultBuilder (IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, FunctionCallSyntaxBase functionCall, ImmutableArray<TypeSymbol> argumentTypes)
+        {
             var arguments = functionCall.Arguments.ToImmutableArray();
-           
-            if (argumentTypes.Length != 1 || argumentTypes[0] is not StringLiteralType stringLiteral)
+            
+            if (argumentTypes.Length < 1 || argumentTypes[0] is not StringLiteralType stringLiteral)
             {
                 return new(ErrorType.Create(DiagnosticBuilder.ForPosition(arguments[0]).CompileTimeConstantRequired()));
             }
             var envVariableName = stringLiteral.RawStringValue;
             var envVariableValue = Environment.GetEnvironmentVariable(envVariableName);
-            
-            //error to fail the build-param with clear message of the missing env var name
+                  
             if (envVariableValue == null)
             {
-                return new(ErrorType.Create(DiagnosticBuilder.ForPosition(arguments[0]).FailedToEvaluateParameter(envVariableName, "Environment variable does not exist")));
+                if (argumentTypes.Length == 2 && argumentTypes[1] is StringLiteralType stringLiteral2)
+                {
+                    return new(TypeFactory.CreateStringLiteralType(stringLiteral2.RawStringValue),
+                new StringLiteralExpression(null, stringLiteral2.RawStringValue));
+                }
+                else
+                {
+                    //error to fail the build-param with clear message of the missing env var name
+                    return new(ErrorType.Create(DiagnosticBuilder.ForPosition(arguments[0]).FailedToEvaluateParameter(envVariableName,
```

Comment:
Is it possible for them to pass in 3 arguments?  If so, they'd get a confusing error message.  Or if they pass in an integer or other non-string value?

Reply:
if they passed more than 2 variables, they get BCP071 error with accurate message around extra arguments
![image](https://github.com/Azure/bicep/assets/34171319/8eb40a4a-42f6-43d2-a687-66aa677e36e9)


if they passed a non string values for any of the arguments they get BCP070 error with accurate message around wrong value type 
![image](https://github.com/Azure/bicep/assets/34171319/01bb1169-a4f8-4eb4-bee0-7c03bbbc3e6b)